internal/charmstore/store.go

```diff
@@ -270,7 +289,7 @@ func (s *Store) insertEntity(entity *mongodoc.Entity) (err error) {
 			Read:  readPerm,
 			Write: writePerm,
 		},
-		Promulgated: entity.PromulgatedURL != nil,
+		Promulgated: promulgated,
```

Comment:
This turned out to be a great idea, it showed up a few potential bugs in the process.